### PR TITLE
Updating credentials and EventData from review

### PR DESCRIPTION
**Pull request by conniey**
* Updates to EventSharedKeyCredential -> EventSharedAccessKeyCredential
  * Removes tokenValidity duration from constructor (this is package-private now)
  * No longer throws checked exceptions.
* Removes EventData implements Comparable
  * Makes getContext() and addContext() package-private
  * Removes addProperty
  * Updates getBody() to return byte[]

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java [hunk 1]
```diff
@@ -93,24 +86,10 @@ public EventData(byte[] body, Context context) {
...
-     * @param body The data to set for this event.
-     * @param context A specified key-value pair of type {@link Context}.
-     * @throws NullPointerException if {@code body} or if {@code context} is {@code null}.
-     */
-    public EventData(ByteBuffer body, Context context) {
-        Objects.requireNonNull(body, "'body' cannot be null.");
-        Objects.requireNonNull(body, "'context' cannot be null.");
-
-        this.body = body;
+        this.body = Objects.requireNonNull(body, "'body' cannot be null.").array();
```
Comment (alzimmermsft) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java]: Should this call into the `byte[]` constructor instead?

```java
this(Objects.requireNonNull(body, "'body' cannot be null.").array());
```

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubSharedKeyCredential.java [hunk 2]
```diff
@@ -82,11 +98,21 @@ public EventHubSharedAccessKeyCredential(String policyName, String sharedAccessK
             throw new IllegalArgumentException("'tokenTimeToLive' has to positive and in the order-of seconds");
         }
 
-        hmac = Mac.getInstance(HASH_ALGORITHM);
+        try {
+            hmac = Mac.getInstance(HASH_ALGORITHM);
+        } catch (NoSuchAlgorithmException e) {
+            throw logger.logExceptionAsError(new AzureException(
```
Comment (alzimmermsft) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubSharedKeyCredential.java]: Is `AzureException` the appropriate exception to be thrown here?

Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubSharedKeyCredential.java]: Any suggestions? Not sure what would apply.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java (hunk 1 above)
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java]: good catch. Fixed

**Pull request merged**